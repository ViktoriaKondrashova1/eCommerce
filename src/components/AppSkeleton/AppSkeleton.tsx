import type { BaseComponent } from '@/shared/types/common.types'
import type { FC } from 'react'
import { Skeleton } from 'antd'
import { Backdrop } from '../Backdrop/Backdrop'

interface Props extends BaseComponent {}

export const AppSkeleton: FC<Props> = ({ testId = 'skeleton' }) => {
  return (
    <Backdrop data-testid={testId} style={{ padding: '16px 0', width: '100%' }}>
      <Skeleton active />
      <Skeleton active />
      <Skeleton active />
      <Skeleton active />
      <Skeleton active />
    </Backdrop>
  )
}
