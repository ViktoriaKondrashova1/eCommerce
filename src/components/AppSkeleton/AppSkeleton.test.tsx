import { render, screen } from '@testing-library/react'
import { describe, expect, it } from 'vitest'
import { AppSkeleton } from './AppSkeleton'
import '@testing-library/jest-dom'

describe('appSkeleton', () => {
  it('should render with default props', () => {
    render(<AppSkeleton />)

    const backdrop = screen.getByTestId('skeleton')
    expect(backdrop).toBeInTheDocument()

    expect(backdrop).toHaveStyle('padding: 16px 0')
    expect(backdrop).toHaveStyle('width: 100%')
  })

  it('should render with correct Skeleton structure', () => {
    const { container } = render(<AppSkeleton />)

    const skeletonElements = container.querySelectorAll('.ant-skeleton')
    expect(skeletonElements.length).toBe(5)

    skeletonElements.forEach((element) => {
      expect(element).toHaveClass('ant-skeleton-active')
    })
  })
})
