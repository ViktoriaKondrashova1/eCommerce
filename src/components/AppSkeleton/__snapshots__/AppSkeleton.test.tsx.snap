// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AppSkeleton > should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="backdrop"
    data-testid="skeleton"
    style="padding: 16px 0px; width: 100%;"
  >
    <div
      class="ant-skeleton ant-skeleton-active css-dev-only-do-not-override-1261szd"
    >
      <div
        class="ant-skeleton-content"
      >
        <h3
          class="ant-skeleton-title"
          style="width: 38%;"
        />
        <ul
          class="ant-skeleton-paragraph"
        >
          <li />
          <li />
          <li
            style="width: 61%;"
          />
        </ul>
      </div>
    </div>
    <div
      class="ant-skeleton ant-skeleton-active css-dev-only-do-not-override-1261szd"
    >
      <div
        class="ant-skeleton-content"
      >
        <h3
          class="ant-skeleton-title"
          style="width: 38%;"
        />
        <ul
          class="ant-skeleton-paragraph"
        >
          <li />
          <li />
          <li
            style="width: 61%;"
          />
        </ul>
      </div>
    </div>
    <div
      class="ant-skeleton ant-skeleton-active css-dev-only-do-not-override-1261szd"
    >
      <div
        class="ant-skeleton-content"
      >
        <h3
          class="ant-skeleton-title"
          style="width: 38%;"
        />
        <ul
          class="ant-skeleton-paragraph"
        >
          <li />
          <li />
          <li
            style="width: 61%;"
          />
        </ul>
      </div>
    </div>
    <div
      class="ant-skeleton ant-skeleton-active css-dev-only-do-not-override-1261szd"
    >
      <div
        class="ant-skeleton-content"
      >
        <h3
          class="ant-skeleton-title"
          style="width: 38%;"
        />
        <ul
          class="ant-skeleton-paragraph"
        >
          <li />
          <li />
          <li
            style="width: 61%;"
          />
        </ul>
      </div>
    </div>
    <div
      class="ant-skeleton ant-skeleton-active css-dev-only-do-not-override-1261szd"
    >
      <div
        class="ant-skeleton-content"
      >
        <h3
          class="ant-skeleton-title"
          style="width: 38%;"
        />
        <ul
          class="ant-skeleton-paragraph"
        >
          <li />
          <li />
          <li
            style="width: 61%;"
          />
        </ul>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`appSkeleton > should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="backdrop"
    data-testid="skeleton"
    style="padding: 16px 0px; width: 100%;"
  >
    <div
      class="ant-skeleton ant-skeleton-active css-dev-only-do-not-override-1261szd"
    >
      <div
        class="ant-skeleton-content"
      >
        <h3
          class="ant-skeleton-title"
          style="width: 38%;"
        />
        <ul
          class="ant-skeleton-paragraph"
        >
          <li />
          <li />
          <li
            style="width: 61%;"
          />
        </ul>
      </div>
    </div>
    <div
      class="ant-skeleton ant-skeleton-active css-dev-only-do-not-override-1261szd"
    >
      <div
        class="ant-skeleton-content"
      >
        <h3
          class="ant-skeleton-title"
          style="width: 38%;"
        />
        <ul
          class="ant-skeleton-paragraph"
        >
          <li />
          <li />
          <li
            style="width: 61%;"
          />
        </ul>
      </div>
    </div>
    <div
      class="ant-skeleton ant-skeleton-active css-dev-only-do-not-override-1261szd"
    >
      <div
        class="ant-skeleton-content"
      >
        <h3
          class="ant-skeleton-title"
          style="width: 38%;"
        />
        <ul
          class="ant-skeleton-paragraph"
        >
          <li />
          <li />
          <li
            style="width: 61%;"
          />
        </ul>
      </div>
    </div>
    <div
      class="ant-skeleton ant-skeleton-active css-dev-only-do-not-override-1261szd"
    >
      <div
        class="ant-skeleton-content"
      >
        <h3
          class="ant-skeleton-title"
          style="width: 38%;"
        />
        <ul
          class="ant-skeleton-paragraph"
        >
          <li />
          <li />
          <li
            style="width: 61%;"
          />
        </ul>
      </div>
    </div>
    <div
      class="ant-skeleton ant-skeleton-active css-dev-only-do-not-override-1261szd"
    >
      <div
        class="ant-skeleton-content"
      >
        <h3
          class="ant-skeleton-title"
          style="width: 38%;"
        />
        <ul
          class="ant-skeleton-paragraph"
        >
          <li />
          <li />
          <li
            style="width: 61%;"
          />
        </ul>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
